extensions/sql-migration/src/api/azure.ts

```
@@ -82,22 +82,44 @@ export async function getBlobContainers(account: azdata.Account, subscription: S
 	return blobContainers!;
 }
 
-export async function getMigrationController(account: azdata.Account, subscription: Subscription, resourceGroupName: string, regionName: string, controllerName: string): Promise<azureResource.MigrationController> {
+export async function getMigrationController(account: azdata.Account, subscription: Subscription, resourceGroupName: string, regionName: string, controllerName: string): Promise<MigrationController> {
 	const api = await getAzureCoreAPI();
-	let result = await api.getMigrationController(account, subscription, resourceGroupName, regionName, controllerName, true);
-	return result.controller!;
+	const host = `https://${regionName}.management.azure.com`;
+	const path = `/subscriptions/${subscription.id}/resourceGroups/${resourceGroupName}/providers/Microsoft.DataMigration/Controllers/${controllerName}?api-version=2020-09-01-preview`;
+	const response = await api.makeAzureRestRequest(account, subscription, path, azurecore.HttpRequestMethod.GET, undefined, true, host);
+	if (response.errors.length > 0) {
+		throw response.errors;
```

You should throw an error - not the raw error array.